client/src/views/auth/Signin.vue

```diff
@@ -8,27 +8,34 @@
 
 <script lang="ts">
 // Libraries
-import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Component, Prop, Mixins } from 'vue-property-decorator'
 
 // Components
 // @ts-ignore
 import SbcSignin from 'sbc-common-components/src/components/SbcSignin'
 
+// Mixins
+import NrAffiliationMixin from '@/components/mixins/nr-affiliation-mixin'
+
 @Component({
   components: {
     SbcSignin
   }
 })
-export default class Signin extends Vue {
+export default class Signin extends Mixins(NrAffiliationMixin) {
   @Prop({ default: 'bcsc' }) idpHint: string
   @Prop({ default: '' }) redirectUrl: string
   @Prop({ default: '' }) redirectUrlLoginFail: string
 
   /** called when keycloak session is ready. */
   private async onReady () {
     if (this.redirectUrl) {
-      // navigate to the route we originally came from
-      await this.$router.push(this.redirectUrl)
+      // navigate to the route we originally came from and process an affiliation if an Nr is pending.
+      await this.$router.push(this.redirectUrl).then(() => {
```

Comment:
What does `then` do on a router push?

Reply:
A promise to resolve something after the redirect.

I believe it could work just the same as 

`await this.$router.push(this.redirectUrl)`
` // If there is stored NR data to process, create the affiliation`
`const nr = JSON.parse(sessionStorage.getItem('NR_DATA'))`
` if (nr) this.createAffiliation(nr)`

Reply:
OK, cool.

Shouldn't the affiliation be created _before_ the route navigation? What if it fails?

Reply:
Another good point. There really is no value in the redirect if we are starting the affiliation process. In this instance, the redirect to root should only happen in the event of a failed affiliation/temp business creation, will refactor this!

Reply:
Since the affiliation process will either result in the user at manage business dashboard or root. This redirect here should only occur when a client is simply logging in through regular means.

Thanks for this!

Reply:
Updated in next commit!

Reply:
Looks good!